render/render.go

```diff
@@ -178,10 +178,12 @@ type Filter struct {
 // Render implements Renderer
 func (f Filter) Render(rpt report.Report) RenderableNodes {
 	output := RenderableNodes{}
+	inDegrees := map[string]int{}
 	for id, node := range f.Renderer.Render(rpt) {
 		if f.FilterFunc(node) {
 			output[id] = node
 		}
+		inDegrees[id] = 0
```

Comment:
Indeed, except if we didn't do the initial set the iteration through all the in degrees below would only yield those with in degree > 0, and we wouldn't filter out those with no incoming edges...